### PR TITLE
Add a new -sil-disable-only-function flag

### PR DESCRIPTION
**Pull request by meg-gupta**
This will enable us apply -sil-disable-pass only on certain functions

### CODE REVIEW

File: lib/SILOptimizer/PassManager/PassManager.cpp [hunk 1]
```diff
@@ -87,6 +87,9 @@ llvm::cl::list<std::string>
     SILDisablePass("sil-disable-pass", llvm::cl::CommaSeparated,
                      llvm::cl::desc("Disable passes "
                                     "which contain a string from this list"));
+llvm::cl::list<std::string> SILDisableOnlyFun(
+    "sil-disable-only-function", llvm::cl::CommaSeparated,
```
Comment (gottesmm) [inline lib/SILOptimizer/PassManager/PassManager.cpp:91]: Can you add pass to the name? Maybe: `-sil-disable-pass-only-function=`. Then it is really clear.

File: lib/SILOptimizer/PassManager/PassManager.cpp [hunk 2]
```diff
@@ -592,7 +601,7 @@ runFunctionPasses(unsigned FromTransIdx, unsigned ToTransIdx) {
 
 void SILPassManager::runModulePass(unsigned TransIdx) {
   auto *SMT = cast<SILModuleTransform>(Transformations[TransIdx]);
-  if (isDisabled(SMT))
+  if (isDisabled(SMT, nullptr))
```
Comment (gottesmm) [inline lib/SILOptimizer/PassManager/PassManager.cpp:604]: Should this be a default nullptr value?

File: lib/SILOptimizer/PassManager/PassManager.cpp [hunk 3]
```diff
@@ -204,7 +207,13 @@ static bool doPrintAfter(SILTransform *T, SILFunction *F, bool Default) {
   return Default;
 }
 
-static bool isDisabled(SILTransform *T) {
+static bool isDisabled(SILTransform *T, SILFunction *F) {
+  if (!SILDisableOnlyFun.empty() && F &&
+      SILDisableOnlyFun.end() == std::find(SILDisableOnlyFun.begin(),
```
Comment (gottesmm) [inline lib/SILOptimizer/PassManager/PassManager.cpp:212]: IIRC, we have a range for std::find, no?

**Review by gottesmm — APPROVED** [review]
Also, can you add a test for this? After that + comments LGTM!

**Comment by meg-gupta** [comment]
@swift-ci smoke test

**Pull request merged**